src/app/services/gateway.service.ts

```diff
@@ -1,25 +1,42 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-import { PostResponse } from 'src/types/posts';
+import { Post } from 'src/types/posts';
+import { Subreddit } from 'src/types/subreddits';
 import { User } from 'src/types/user';
 
 @Injectable({
     providedIn: 'root',
 })
 export class GatewayService {
-    private _data = 'assets/posts.json';
+    private _mockPostData = 'assets/posts.json';
+    private _mockSubredditData = 'assets/subreddits.json';
+
+    private _baseUrl = 'http://localhost:3000/api/v1/';
```

Comment:
The comments in this file are not saying much. 
```
// subreddits
function fetchSubreddits() {...}
```

You make a comment where the function itself already mentions it. It should not be in there.

Furthermore, the API names are odd. You can't really use base url like this, especially because it will be different for different environments. This may already be solved with heroku deployment. 

Similar to previous comment, `_feedData` and `baseUrl/feed` are telling exactly the same. It does not have to be a property.

Reply:
Ok, I have deleted the properties and comments. The baseURL stayed the same, because Martin will change it with heroku deployment.